### DISCUSSION TITLE
[09 - PyTorch Model Deployment] Hugging Face URL Error for DEFAULT weights

### DISCUSSION THREAD

**Discussion by Bruno-Messias** [mrdbourke/pytorch-deep-learning#272] [Q&A] [answered]
Hello Everyone!

Today I was trying to deploy the food vision model to Hugging Face, but I come up to a **URL Error** at hugging face logs that I dont understood. The error was showing me that the download link of PyTorch is not working for the `DEFAULT ` weights of `EfficientNet_B2_Weights`. I manage to make work changing to `IMAGENET1K_V1` weights, like that:

```python
weights = torchvision.models.EfficientNet_B2_Weights.IMAGENET1K_V1
```
But nothing like that happened when I was coding at Google Colab.
 
Does anyone have an idea of what it could be? Or come with the same error?

**Comment by mrdbourke** [comment]
Hey hey @Bruno-Messias,

I'm not what could be happening here.

Looks like from the PyTorch documentation that `.DEFAULT` should work fine?

See: https://pytorch.org/vision/stable/models/generated/torchvision.models.efficientnet_b2.html#torchvision.models.EfficientNet_B2_Weights

```
The model builder above accepts the following values as the weights parameter. 

EfficientNet_B2_Weights.DEFAULT is equivalent to EfficientNet_B2_Weights.IMAGENET1K_V1. You can also use strings, e.g. weights='DEFAULT' or weights='IMAGENET1K_V1'.

These weights are ported from the original paper. Also available as EfficientNet_B2_Weights.DEFAULT.
```

But this error is only happening to you on Hugging Face not Colab?

**Reply by Bruno-Messias** [reply]
It's only happening in Hugging Face. On Colab all works fine.

**Reply by mrdbourke** [reply]
🤔 

That is strange to me

Potentially Hugging Face are using a different version of PyTorch?

Does the Gradio app still run?

**Comment by Bruno-Messias** [comment, marked as answer]
I think I cath the error! Thanks the insigth!

The requirements was something different from the course, I was using this as requirement:
```
torch==1.13.0
torchvision==0.14.0
```
But at the course there are created as:
```
torch==1.12.0
torchvision==0.13.0
```

I just changed these requirements and the weights  to ` DEFAULT ` and works!
